### ISSUE TITLE
Difference between Docker machine "name" vs DNS name.

### ISSUE THREAD

**Issue by kolotyluk** [docker/docker.github.io#4379]
File: [machine/concepts.md](https://docs.docker.com/machine/concepts/), CC @londoncalling 

Generally the documentation for docker-machine is just as abysmal as the the User Interface for it. I am sure for someone working at Docker is all makes perfect sense, but for someone new to docker-machine it makes no sense at all.

For example, I have created a swarm on AWS using the Docker Swarm CloudFormation template. I have opened up the security groups so that I can ssh into my nodes from my computer. However, if I use something like

`docker-machine ssh . . .`

I get back an utterly useless message

`Host does not exist: ". . ."`

So what exactly is a docker-machine `machine`?

If I can ssh to some dns name, then `docker-machine ssh <dns-name>` should actually do something useful.

**Comment by londoncalling** [comment]
Hi @kolotyluk , per the login step shown on the reference page [docker-machine ssh](https://docs.docker.com/machine/reference/ssh/), you need to provide the name of your virtual machine as an argument to `docker-machine ssh`. In the example you provided above, you are using `. . .` as a machine name, which I don't think you intended.

Here is an example I just ran on my laptop:

1. I create two machines called `my-vm1` and `my-vm2` using the commands `docker-machine create --driver virtualbox my-vm1` and `docker-machine create --driver virtualbox my-vm2`. Creating these one at a time, takes a minute, and you'll get output back indicating when each is created.

2. I check to see that the machines are there:

```
$ docker-machine ls
NAME     ACTIVE   DRIVER       STATE     URL                         SWARM   DOCKER        ERRORS
my-vm1   -        virtualbox   Running   tcp://192.168.99.100:2376           v17.06.1-ce   
my-vm2   -        virtualbox   Running   tcp://192.168.99.101:2376           v17.06.1-ce   
```

3. Now I can log on to these machines in one of two ways. I can use `ssh`, or I can get the environment variables for one or the other of them. Since you are trying to log in using `ssh`, here is how you would log in to `my-vm`:

```
victoriabialas at snapair in ~
$ docker-machine ssh my-vm1
                        ##         .
                  ## ## ##        ==
               ## ## ## ## ##    ===
           /"""""""""""""""""\___/ ===
      ~~~ {~~ ~~~~ ~~~ ~~~~ ~~~ ~ /  ===- ~~~
           \______ o           __/
             \    \         __/
              \____\_______/
 _                 _   ____     _            _
| |__   ___   ___ | |_|___ \ __| | ___   ___| | _____ _ __
| '_ \ / _ \ / _ \| __| __) / _` |/ _ \ / __| |/ / _ \ '__|
| |_) | (_) | (_) | |_ / __/ (_| | (_) | (__|   <  __/ |
|_.__/ \___/ \___/ \__|_____\__,_|\___/ \___|_|\_\___|_|
Boot2Docker version 17.06.1-ce, build HEAD : 80114bc - Fri Aug 18 17:58:04 UTC 2017
Docker version 17.06.1-ce, build 874a737

```

Now I'm in. When you want to log out, just type `exit`.

4. To log in to `my-vm2`, use the same command, but this time provide that machine name:

```
victoriabialas at snapair in ~
$ docker-machine ssh my-vm2
                        ##         .
                  ## ## ##        ==
               ## ## ## ## ##    ===
           /"""""""""""""""""\___/ ===
      ~~~ {~~ ~~~~ ~~~ ~~~~ ~~~ ~ /  ===- ~~~
           \______ o           __/
             \    \         __/
              \____\_______/
 _                 _   ____     _            _
| |__   ___   ___ | |_|___ \ __| | ___   ___| | _____ _ __
| '_ \ / _ \ / _ \| __| __) / _` |/ _ \ / __| |/ / _ \ '__|
| |_) | (_) | (_) | |_ / __/ (_| | (_) | (__|   <  __/ |
|_.__/ \___/ \___/ \__|_____\__,_|\___/ \___|_|\_\___|_|
Boot2Docker version 17.06.1-ce, build HEAD : 80114bc - Fri Aug 18 17:58:04 UTC 2017
Docker version 17.06.1-ce, build 874a737
```

I can type some commands:
```
docker@my-vm2:~$ ls
log.log
docker@my-vm2:~$ pwd
/home/docker
```
Then exit:
```
docker@my-vm2:~$ exit
```
The primary use case for Docker Machine at this point is for provisioning multiple hosts locally, and the benefit it provides is that you don't have to use `ssh` if you don't want to. You run `eval "$(docker-machine env <machine-name)`; e.g., `eval "$(docker-machine env my-vm1)` per our examples, and your terminal will be talking to that machine without doing an `ssh`.

For information on using the other set of commands (not `ssh`) see the topics on how to [create a machine ](https://docs.docker.com/machine/get-started/#create-a-machine) and [Run containers and experiment with Machine commands](https://docs.docker.com/machine/get-started/#run-containers-and-experiment-with-machine-commands).

Please provide links for topics that were not useful to you or where you were getting frustrated.

Thanks for the feedback.

Vicky

**Comment by johndmulhausen** [comment]
@kolotyluk Do you have any response to @londoncalling's feedback? After no answer for the last few days I'm inclined to close your bug. Note: For future reference, it is best to give bugs a more helpful and specific title, and in the body of your bug, provide actionable feedback.

**Comment by kolotyluk** [comment]
I tried responding via e-mail, but I guess that does not work.

To summarize, if I type
```
docker-machine ssh <dns-name>
```
it responds with
```
Host does not exist: "<dns-name>"
```
A less ambiguous response would be
```
Docker container "<dns-name>" does not exist.
```

**Comment by londoncalling** [comment]
@kolotyluk I can chat with the Docker Machine dev's about the output message, but also I think there is confusion about what a "Docker machine" is, in that it's a specific type of virtual machine that you create with the `docker-machine` command.  That is on us both in the documentation and in the concept of the product itself. 

I can see where a person would think that `docker-machine ...` followed by some set of arguments might apply to any host. We've tried to clarify this in the docs, but I'll have a look back through to see if there is more I can improve, and also check out what we can do regarding the command line output message you mention. 

Thanks,
Vicky

**Comment by londoncalling** [comment]
@kolotyluk Thanks for your feedback on this, you helped make the CLI help better.

This is fixed per PR #4237
